projects/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-top-posts.php

```diff
@@ -0,0 +1,201 @@
+					},
+				),
+			)
+		);
+
+		// Number of posts and selected post types are not needed in the Editor.
+		// This is to minimise requests when it can already be handled by the block.
+		register_rest_route(
+			'wpcom/v2',
+			'/top-posts',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_top_posts' ),
+					'permission_callback' => '__return_true',
+					'args'                => array(
+						'period' => array(
+							'description'       => __( 'Timeframe for stats.', 'jetpack' ),
+							'type'              => array( 'string', 'integer' ),
+							'required'          => true,
+							'validate_callback' => function ( $param ) {
+								return is_numeric( $param ) || is_string( $param );
+							},
+						),
+						'number' => array(
+							'description'       => __( 'Number of posts to display.', 'jetpack' ),
+							'type'              => 'integer',
+							'required'          => false,
+							'validate_callback' => function ( $param ) {
+								return is_numeric( $param );
+							},
+						),
+						'types'  => array(
+							'description'       => __( 'Types of content to include.', 'jetpack' ),
+							'type'              => 'string',
+							'required'          => false,
+							'validate_callback' => function ( $param ) {
+								return is_string( $param );
+							},
+						),
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Get the site's post types.
+	 *
+	 * @return array Site's post types.
+	 */
+	public function get_post_types() {
+		$post_types       = array_values( get_post_types( array( 'public' => true ) ) );
+		$post_types_array = array();
+
+		foreach ( $post_types as $type ) {
+			$post_types_array[] = array(
+				'label' => get_post_type_object( $type )->labels->name,
+				'id'    => $type,
+			);
+		}
+
+		return $post_types_array;
+	}
+
+	/**
+	 * Get the site's top content.
+	 *
+	 * @param \WP_REST_Request $request request object.
+	 *
+	 * @return array Data on top posts.
+	 */
+	public function get_top_posts( $request ) {
+		$period        = $request->get_param( 'period' );
+		$all_time_days = floor( ( time() - strtotime( get_option( 'site_created_date' ) ) ) / ( 60 * 60 * 24 * 365 ) );
+
+		// While we only display ten posts, users can filter out content types.
+		// As such, we should obtain a few spare posts from the Stats endpoint.
+		$posts_to_obtain_count = 30;
+
+		// We should not override cache when displaying the block on the frontend.
+		// But we should allow instant preview of changes when editing the block.
+		$is_rendering_block = isset( $request['types'] );
+		$override_cache     = ! $is_rendering_block;
+
+		$query_args = array(
+			'max'       => $posts_to_obtain_count,
+			'summarize' => true,
+			'num'       => $period !== 'all-time' ? $period : $all_time_days,
+			'period'    => 'day',
+		);
+
+		$data            = ( new WPCOM_Stats() )->get_top_posts( $query_args, $override_cache );
```

Comment:
Hi @jeherve,

Just in case you have some insights. When I wired up the endpoint on WPCOM, the call constantly fails. I had look. On WPCOM, the API call was rerouted to use `WPCOM_API_Direct::do_request()`, and then fails with an error: 
```
{"code":"invalid_namespace","message":"Invalid API namespace","data":{"status":404}}
```

I searched the error message, which seems to be thrown by `wp-content/rest-api-plugins/centralized/centralize.php` - `wpcom/v2`. I'm wondering why `v1` endpoints couldn't be found 🤔

Reply:
> I'm wondering why v1 endpoints couldn't be found 🤔

Since the diff is outdated I'm not sure I fully understand, but I don't believe we ever had `wpcom/v1` endpoints in prod.